src/ImportDefinitionsBundle/Interpreter/AssetUrlInterpreter.php

```diff
@@ -56,8 +58,21 @@ public function interpret(Concrete $object, $value, Mapping $map, $data, Definit
             $context = new PlaceholderContext($data, $object);
             $assetPath = $this->placeholderService->replace($path, $context);
             $assetFullPath = sprintf('%s/%s', $assetPath, $filename);
+            
+            if ($configuration['deduplicate_by_url']) {
+                $listing = new Asset\Listing();
+                $listing->onCreateQuery(function (\Pimcore\Db\ZendCompatibility\QueryBuilder $select){
+                    $select->join('assets_metadata AS am', 'id = am.cid', ['cid']);
+                });
+                $listing->addConditionParam('am.name = ?', self::METADATA_ORIGIN_URL);
+                $listing->addConditionParam('am.data = ?', $value);
+                $listing->setLimit(1);
```

Comment:
This avoids a possible problem where you turn the setting on and off and on and off and get multiple matches, it just takes the first match.